lib/node.js

```diff
@@ -502,6 +502,40 @@ class Node {
     return node.canReplaceWith(this)
   }
 
+  matches (node) {
+    // if the nodes are literally the same object, obviously a match.
+    if (node === this)
+      return true
+
+    // if the names don't match, they're different things, even if
+    // the package contents are identical.
+    if (node.name !== this.name)
+      return false
+
+    // if they're links, they match if the targets match
+    if (this.isLink)
+      return node.isLink && this.target.matches(node.target)
+
+    // if they're two root nodes, they're different if the paths differ
+    if (this.isRoot && node.isRoot)
+      return this.path === node.path
```

Comment:
shouldn't it also check for `realpath` in order to cover the root-is-a-link usecase?

Reply:
Hmm... probably right.

Reply:
Oh!  No, not relevant, because the `node.isLink` check above ensures that they'd only be a match if they're both links and the targets match.